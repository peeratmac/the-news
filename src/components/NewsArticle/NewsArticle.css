.news-article {
  display: flex;
  flex-direction: column;
  align-items: center;
  border-bottom: 2px solid #ff7979;
}

img {
  margin: 20px;
  border-radius: 20px;
  height: 300px;
  object-fit: cover;
  width: 100%;
}

.link {
  margin-top: 10px;
  text-decoration: none;
  background: #f6e58d;
  padding: 10px;
  border-radius: 5px;
  margin-bottom: 15px;
}

.link:hover {
  background: #f39c12;
}

.link-to-article:hover {
  color: #130f40;
}

.headline {
  background: #f6e58d;
  text-decoration: none;
  padding: 10px;
  border-radius: 10px;
}

.headline:hover {
  background: #ff7979;
}

h2 {
  color: #ff7979;
}

h3 {
  font-size: 20px;
  color: #130f40;
}

p {
  height: 100px;
  overflow: scroll;
}

.description-text {
  color: #130f40;
  font-size: 18px;
}
