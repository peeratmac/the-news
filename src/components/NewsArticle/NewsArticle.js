import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({ description, headline, image, url }) => {
  return (
    <div className='news-article'>
      <a href={url}>
        <h3>{headline}</h3>
      </a>
      <img src={image} alt={headline} />
      <p className='description-text'>{description}</p>
      <a href={url}>
        <h2 className='link-to-article'>Link to Article</h2>
      </a>
    </div>
  );
};

export default NewsArticle;
