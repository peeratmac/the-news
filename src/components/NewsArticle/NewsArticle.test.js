import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should match the NewsArticle Snapshot', () => {
    const wrapper = shallow(
      <NewsArticle
        key={5}
        headline='The Chainsmokers are playing in Denver tonight!'
        image='https://github.com/peeratmac'
        description='Push My Luck, Closer, Paris, You Owe Me, Great Songs'
        url='https://www.linkedin.com/in/peerats'
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
