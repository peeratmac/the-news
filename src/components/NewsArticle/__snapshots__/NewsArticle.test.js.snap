// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the NewsArticle Snapshot 1`] = `
<div
  className="news-article"
>
  <a
    href="https://www.linkedin.com/in/peerats"
  >
    <h3>
      The Chainsmokers are playing in Denver tonight!
    </h3>
  </a>
  <img
    alt="The Chainsmokers are playing in Denver tonight!"
    src="https://github.com/peeratmac"
  />
  <p
    className="description-text"
  >
    Push My Luck, Closer, Paris, You Owe Me, Great Songs
  </p>
  <a
    href="https://www.linkedin.com/in/peerats"
  >
    <h2
      className="link-to-article"
    >
      Link to Article
    </h2>
  </a>
</div>
`;
